### ISSUE TITLE
[BUG]  Chia does not work with python 3.10

### ISSUE THREAD

**Issue by moonlitbugs** [Chia-Network/chia-blockchain#8807]
**Describe the bug**

After a routine OS upgrade from Python 3.9 to 3.10, ```chia start node```  fails to start the node.  While the **chia_daemon** process starts (and writes out its ```mainnet/run/start-daemon.launching```), the main "*chia*" CLI thinks that it has not done so, and refuses to start any of the other requested processes (node, farmer, etc).

**To Reproduce**

1. Upgrade OS from Python 3.9 to 3.10
2. (re)Build Chia
```
~$ cd /usr/local/chia-blockchain
~$ rm -rf venv
~$ bash ./install.sh
...
~$ . ./activate
(venv) ~$ chia version
1.2.9
(venv) ~$ chia start node
Daemon not started yet
Starting daemon
Daemon not started yet
Failed to create the chia daemon
(venv) ~$ pgrep chia
4368
(venv) ~$ ps -p 4368
  PID TTY          TIME CMD
 4368 pts/0    00:00:01 chia_daemon
(venv) ~$ cat ../mainnet/log/debug.log
2021-10-12T17:22:24.578 daemon chia.daemon.server         : INFO     Starting Daemon Server
2021-10-12T17:22:24.580 daemon chia.daemon.server         : INFO     Waiting Daemon WebSocketSer
(venv) $~ chia stop -d node
Daemon not started yet
Couldn't connect to chia daemon
(venv) ~$ killall chia_daemon
(venv) ~$ cat ../mainnet/log/debug.log
2021-10-12T17:22:24.578 daemon chia.daemon.server         : INFO     Starting Daemon Server
2021-10-12T17:22:24.580 daemon chia.daemon.server         : INFO     Waiting Daemon WebSocketServer closure
2021-10-12T17:39:34.066 daemon chia.daemon.server         : INFO     chia daemon exiting in 5 seconds
(venv) ~$
```
3. Build an OS package for Python3.9 that installs to /opt or somewhere other than /usr.
   Install it.  [ Insert expletives here ]
4. Rebuild Chia
```
~$ python --version
Python 3.9.7
~$ cd /usr/local/chia-blockchain
~$ rm -rf venv
~$ bash ./install.sh
...
~$ . ./activate
(venv) ~$ chia version
1.2.9
(venv) ~$ chia start farmer
Daemon not started yet
Starting daemon
chia_harvester: Already running, use `-r` to restart
chia_farmer: Already running, use `-r` to restart
chia_full_node: Already running, use `-r` to restart
chia_wallet: Already running, use `-r` to restart
(venv) ~$
```

**Expected behavior**

Chia should start normally, as in the latter half of the description, above.

**Desktop**

- OS: *Slackware* Linux
- OS Version/Flavor: *Current* (pre-15.0)
- CPU: x86_64, Ryzen 2700

**Additional context**

Note that the GUI works just fine under Python 3.10.  Only the daemon has issues.  (I didn't try the timelord, as the CPU running the actual daemon - *not my development machine* - is much older.)

**Comment by moonlitbugs** [comment]
Not sure if the devos have python 3.10 on their machines.  If not, and the maintainer of the daemon code wants to throw me a patch, I'd be happy to test.  I've eyeballed the code briefly, but didn't see anything that stood out.  Getting python's regular *stderr* output visible might be a win here; but debugging python under Chia is "difficult" for anything not going to mainnet/log/.

**Comment by paulhainsworth-chia** [comment]
@danieljperry if python 3.10 is not supported, we need to add this to our documentation to specifically flag that it's not supported. Prob need to add to our backlog to start supporting 3.10 shortly.

**Comment by paulhainsworth-chia** [comment]
Added to backlog: https://trello.com/c/2h3D7bil

**Comment by danieljperry** [comment]
Noted.

**Comment by Pencerval** [comment]
Same issue here on Clear Linux. Python 3.19 doesn't come with it anymore.

**Comment by Ealrann** [comment]
Same here on Archlinux. Python 3.10 is now the default, and the chia daemon fail to start :
```
Failed to create the chia daemon
```

**Comment by SemnoSao** [comment]
I'm still getting this same error. Is it solved already? Did I miss something? Just upgraded my manjaro and chia and can't open the gui anymore. Trying to start by the CLI gives the same error mentioned above.

**Comment by hoffmang9** [comment]
The Chia daemon will not run under Python 3.10.

One workaround option is to install a python 3.9 locally and hand create a venv and install chia there.
````
cd chia-blockchain
/usr/local/path-to/python39 -m venv venv
. ./bin/activate
sh install.sh
```

**Comment by SemnoSao** [comment]
did exactly those steps and still got the error:
![image](https://user-images.githubusercontent.com/16153554/148840266-87c72526-fe5b-4ecc-a0ce-0ebcad1ce8f9.png)
I have run the install.sh before as instructed and attempted to start farmer after but got the same error. I uploaded the picture so anyone can call out if I missed any step or did anything wrong.

**Comment by hoffmang9** [comment]
Two things - one I forgot to mention, one you need to check.

You will want to `rm -rf venv` before creating it so
```
cd chia-blockchain
rm -rf venv
/usr/local/path-to/python39 -m venv venv
. ./bin/activate
sh install.sh
```
Also, double check that you don't have a zombie Chia Daemon running as that's easy to do when trying under 3.10.

**Comment by SemnoSao** [comment]
Perfect. Everything is working just fine now, thank you!

**Comment by emlowe** [comment]
Targeted to be fixed in 1.3.5

**Comment by shawnsmithdev** [comment]
As far as I can tell there is no reasonable way to get python 3.9 from either apt or snap in Ubuntu 22.04, so my farm is offline for now.  No big deal; this is exactly one should actually, you know, test things will work before just doing OS upgrades in production.  Woops.

But I am keeping an eye out for 1.3.5 now for sure. Thanks for working on this :)

**Comment by benthetechguy** [comment]
If you're comfortable adding a PPA, there's one for older versions of Python:
```
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt install python3.9
```
Then simply edit install.sh to use python3.9 instead of python3 for venv creation.

<sub> P.S. I'd recommend using Debian stable instead of Ubuntu for things like this </sub>

**Comment by emlowe** [comment]
Note, main right now does work with python 3.10

**Comment by lionel-lbr** [comment]
I solved this issue (at least for harvester, i didn't test on full node) on 1.3.4 by installing Python 3.9.11 manually from source, alongside with 3.10.  Here is how i did it:
1. I started with Ubuntu 22 and Python 3.10 already installed.
2. update build environment for Python with:
```
sudo apt-get update; sudo apt-get install make build-essential libssl-dev zlib1g-dev \
libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
```  
3. download Python 3.9.11, untar and cd into the folder:
```
wget https://www.python.org/ftp/python/3.9.12/Python-3.9.12.tgz
tar -xvzf  Python-3.9.12.tgz
cd Python-3.9.12.tgz
``` 
4. Configure, build and install:
```
./configure --enable-optimizations --enable-shared
make 
```
_Note that you can also run `make -j <number of cpu>`_ 

5. install in /usr/local:
```
sudo make install 
```
6. run `ldconfig` to bind `python3` to /usr/local/bin/python3.9`
```
sudo ldconfig $PWD
```

Python 3.9 should now be installed and running `python3 --version` should show: `Python 3.9.11`.  But you can still run 3.10 with `python3.10`

If chia-blockchain is already checked-out, delete the venv folder: ` rm -rf chia-blockchain/venv`, then edit `install.sh` and comment out line 129 and add line 130 as shown below:
```
127    echo "Installing on Ubuntu 21.04 or newer."
128    sudo apt-get update
129#    sudo apt-get install -y python3.9-venv python3-distutils openssl
130    sudo apt-get install -y openssl
``` 

Save the file and run: `sh install.sh`, the install should show: `Python version is 3.9.11` and should complete successfully.

**Comment by emlowe** [comment]
Fixed in 1.3.5
